source/frontend-docs.rst

```diff
@@ -0,0 +1,218 @@
+
+In your config file, you can set up the source and destination directories
+that JSDoc will look through as it crawls your JS code and builds the documentation
+site. In projects created with our project template, the source dir will look
+like ``project_name/static/js``; we recommend ``docs/js`` as the output dir.
+You'll also want to enable the ``recurse`` option so that your entire source tree
+will be scanned. If your JavaScript build output is located in the same directory
+as your source, you will also want to throw in an ``excludePattern``. Including
+these gives you a config file like::
+
+    {
+      "source": {
+        "excludePattern": "bundle\\.js",
+        "include": [
+          "project_name/static/js"
+        ]
+      },
+      "opts": {
+        "recurse": true,
+        "destination": "docs/js"
+      }
+    }
+
+If your project is based off the Caktus project template, it probably uses
+React as well as the JSX component-specification DSL; and it probably also
+uses Babel to preprocess both. Because JSDoc can use your existing Babel
+setup, it takes very little additional work to ensure that JSDoc can read
+your code. You will need the ``jsdoc-babel`` plugin::
+
+    npm install --save-dev jsdoc-babel
+
+And you will need to add lines to your config file to add the plugin and to
+get JSDoc to recognize ``.jsx`` files as well as ``.js`` ones::
+
+    {
+      "plugins": ["node_modules/jsdoc-babel"],
+      "source": {
+        "includePattern": ".+\\.jsx?$",
+        /* ... */
+      },
+      /* ... */
+    }
+
+That's it. Assuming Babel is correctly configured to process your JSX code,
+JSDoc will now be able to handle it, too.
+
+Once the configuration file is set up, you can generate a JSDoc site from your
+JS by running this command (assuming ``./node_modules/.bin`` is in your
+``$PATH``)::
+
+    jsdoc -c .jsdoc.json
+
+That's it! Your JSDoc documentation will have been deposited in ``./docs/js``.
+
+JSDoc is not a library and doesn't expose a full-featured API. For that
+reason, to use JSDoc from within your build process, it should be treated as
+just another command line program to be run using Node's ``child_process``.
+
+JSDoc: suggested comment style
+==============================
+
+JSDoc offers a very rich selection of tags and other features and a great deal
+of flexibility in how it can be used. We recommend adhering to at least these
+conventions:
+
+* Create a doc comment for every module, class, function / method, and event.
+* For every function / method doc comment, include:
+    * A prose description of the function's purpose
+    * ``@param`` tags each expected parameter, including:
+        * name
+        * type
+        * prose description
+    * For Object-type parameters, include separate ``@param`` entries for each
+      property expected on the Object
+    * ``@returns`` tag giving the function's return value, including:
+        * type
+        * prose description
```

Comment:
Should we include doc on the properties that will be included on a returned Object, like we do for incoming parameters?